src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5956,8 +5956,33 @@ private bool IsStartOfTypeParameter()
                 return false;
 
             // possible attributes
-            if (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken && this.PeekToken(1).Kind != SyntaxKind.CloseBracketToken)
-                return true;
+            if (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken)
+            {
```

Comment:
i would also be fine simplifying this so that we always return true when we see `[` regardless of what follows.